### PR TITLE
Tables!

### PR DESCRIPTION
**Pull request by KimberlyMunoz**
Adding tables!

oh-how-the-tables-have-turned-o
 

## Additions

- Added cf-tables
- Added macro for table creation

## Removals

- Removed old conflicting table styles 

## Testing

- Visit /sublanding-page-2/ and any other tables we might have to revisit

## Review

- @anselmbradford 
- @jimmynotjim 
- @sebworks 

## Preview

image

image

[Preview this PR without the whitespace changes](?w=0)

## Notes

tumblr_m3f00uzxkl1qia9wro2_500

### CODE REVIEW

File: cfgov/jinja2/v1/_includes/organisms/table.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
+
+{# ==========================================================================
+
+   table.render()
+
+   ==========================================================================
+
+   Description:
+
+   Create a gray well organism.
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: Update or deleto

File: cfgov/jinja2/v1/_includes/organisms/table.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+   See [GHE]/flapjack/Modules-V1/wiki/Well
+
+   table.row
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: Needs `thead` and `rows` description.

File: cfgov/jinja2/v1/_includes/organisms/table.html (hunk 1 above)
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: Note to self: just update all the documentation.

File: cfgov/jinja2/v1/_includes/organisms/table.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+
+   ========================================================================== #}
+
+{% macro render(thead, rows) %}
+<table class="table__stack-on-small table__entry-header-on-small">
```
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: Is there a top-level `table` class?

Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: Yes, but no styles are included and it's not needed.

File: cfgov/unprocessed/css/cf-enhancements.less [hunk 4]
```diff
@@ -330,6 +249,8 @@ tbody {
     - Styles first td as if it were the thead
     - Allows any other td to be styled as a heading level 4
     - Adds data-label attr for small td labels above td content
+    - Used in older blog posts to make them responsive.
+      Do not remove until removed from blog posts.
```
Comment (anselmbradford) [inline cfgov/unprocessed/css/cf-enhancements.less]: Add `TODO` comment?

File: cfgov/jinja2/v1/_includes/organisms/table.html (hunk 3 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: There was a thread on cf-typography about `list` (which has no styles), but I believe Mike and Scott said still needs to be included. This is an analogous use-case. Maybe @Scotchester remembers the context?

File: cfgov/jinja2/v1/_includes/organisms/table.html [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            {% endfor %}
+        </tr>
+    </thead>
+
+        {% for row in rows %}
+            {% set colcount = 0 %}
+            <tr>
+                {% for cell in row %}
+                    <td data-label="{{ thead[colcount] }}">{{ cell }}</td>
+                    {% set colcount = colcount + 1 %}
```
Comment (sebworks) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: could you just use `loop.index0` instead?

File: cfgov/jinja2/v1/_includes/organisms/table.html (hunk 3 above)
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: It's not included in [cf-tables](https://cfpb.github.io/cf-tables/demo/) and I think something like this on BEM style should be included in our front-end standards if we have a standard.

File: cfgov/jinja2/v1/_includes/organisms/table.html (hunk 5 above)
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: Lemme try that.

File: cfgov/jinja2/v1/_includes/organisms/table.html (hunk 3 above)
Comment (Scotchester) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: As mentioned on the `list` topic, it's not actually needed. It's a "ghost block" of sorts, that only serves to be the basis for the necessary modifiers.

@KimberlyMunoz I agree, it'd be good to mention this scenario [where we talk about our BEM style](https://github.com/cfpb/front-end/blob/master/css.md#naming-conventions)

File: cfgov/jinja2/v1/_includes/organisms/table.html (hunk 5 above)
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: Yep. Much cleaner! Thanks!

**Comment by sebworks** [comment]
:+1:, just wondering if there should be a full-width modifier for the table.

File: cfgov/jinja2/v1/_includes/organisms/table.html [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+<table class="table__stack-on-small table__entry-header-on-small">
+    <thead>
+        <tr>
+            {% for cell in thead %}
+                <th>{{ cell }}</th>
+            {% endfor %}
+        </tr>
+    </thead>
+
+        {% for row in rows %}
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: This whole for loop doesn't need the extra indentation.

File: cfgov/jinja2/v1/sublanding-page-2/index.html [hunk 7]
```diff
@@ -49,7 +50,16 @@
                          'tty':    '1234567899'}]
         }) }}
     </div>
-
+    <div class="block">
+        {{ table.render( [
+                            'Column 1', 'Column 2', 'Column 3'
+                          ], [
+                          ['Cell A1', 'Cell A2', 'Cell A3'],
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/sublanding-page-2/index.html]: These three inner arrays should be indented once more.

File: cfgov/unprocessed/css/cf-enhancements.less [hunk 8]
```diff
@@ -227,100 +227,19 @@
...
-    th {
-        .webfont-medium();
-    }
-
-    tr {
-        border-bottom: 1px solid @tr-border;
-    }
-}
 
 .respond-to-max(960px, {
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/cf-enhancements.less]: Should this be reduced to `@bp-sm-max`?

**Comment by jimmynotjim** [comment]
Minor comments but overall looks good. 

Side note, I think @KimberlyMunoz wins the __CFPB PR Gifslinger award__

File: cfgov/jinja2/v1/_includes/organisms/table.html (hunk 3 above)
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/_includes/organisms/table.html]: Added an issue on https://github.com/cfpb/front-end/issues/98

**Comment by KimberlyMunoz** [comment]
> Just wondering if there should be a full-width modifier for the table.

I think for mobile styles, we probably want the full-width to be the default. I'll open a PR on cf-tables for that.

**Comment by jimmynotjim** [comment]
:+1:

**Comment by sebworks** [comment]
:+1:

**Pull request merged**